.github/workflows/build.yaml

```diff
@@ -0,0 +1,65 @@
+name: build
+
+on:
+  push:
+    branches:
+      - "branch-*"
+    tags:
+      - v[0-9][0-9].[0-9][0-9].[0-9][0-9]
+  workflow_dispatch:
+    inputs:
+      branch:
+        required: true
+        type: string
+      date:
+        required: true
+        type: string
+      sha:
+        required: true
+        type: string
+      build_type:
+        type: string
+        default: nightly
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  cpp-build:
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/conda-cpp-build.yaml@cuda-118
```

Comment:
update branches to `branch-23.02`

see here: https://github.com/rapidsai/cudf/blob/d1cebdbf50d92358962bc66a1b2c0a8476de1f35/.github/workflows/build.yaml#LL31C83-L31C95